rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala

```diff
@@ -418,7 +425,37 @@ class RSpace[F[_], C, P, A, R, K] private[rspace] (
       } yield result
     }
 
-  override def createCheckpoint(): F[Checkpoint] = ???
+  private def createCache: F[Cell[F, Cache[C, P, A, K]]] =
+    Cell.refCell[F, Cache[C, P, A, K]](Cache())
+
+  private def createNewHotStore(historyReader: HistoryReader[F, C, P, A, K]): F[Unit] =
+    for {
+      cache <- createCache
+      nextHotStore = {
+        implicit val C: Cell[F, Cache[C, P, A, K]]    = cache
+        implicit val HR: HistoryReader[F, C, P, A, K] = historyReader
+        HotStore.inMem
```

Comment:
The implicits are quite verbose. Can't you just call HotStore.inMem with the expected parameters?